typed-racket-lib/typed-racket/types/subtype.rkt

```diff
@@ -374,6 +372,20 @@
 	       (subtype* t1 t2)
 	       (subtype* t2 t1)))
 
+(define union-super-cache (make-weak-hasheq))
+(define union-sub-cache (make-weak-hasheq))
+
+(define/cond-contract (cache-set! cache t1 t2 result)
+  (-> hash? Type? Type? boolean? void?)
+  (hash-set! (hash-ref cache t1 (λ () (make-weak-hash))) t2 (box-immutable result)))
+
+(define/cond-contract (cache-ref cache t1 t2)
+  (-> hash? Type? Type? (or/c #f (box/c boolean?)))
```

Comment:
No. I'll add a comment. A box returned means we have a cached answer, yes or no (a `boolean?`). Returning `#f` directly means we don't have a cached answer.